### PR TITLE
[4.1] use web asset manager

### PR DESCRIPTION
**Pull request by brianteeman**
First pass for files already using $wa to make sure all scripts are loaded with the web asset manager

code review

### CODE REVIEW

**Comment by PhilETaylor** [comment]
The indentation of the chained method calls is wrong in almost every file modified and should be corrected.

**Comment by brianteeman** [comment]
Can you give me an example please - as the current core was kind of hard to work out how it should be

**Comment by brianteeman** [comment]
did you mean that this
```
$wa	->useScript('multiselect')
	->useScript('com_installer.changelog');
```

should be
```
$wa->useScript('multiselect')
	->useScript('com_installer.changelog');

```

**Comment by PhilETaylor** [comment]
Well once again we reach a corner of the decade old Joomla Coding Standards that doesn't take modern code into account and there is no defined example in the standard (hence why phpcs passes in drone) however there are some examples (that contradict each other) in the last 2 code blocks on the https://developer.joomla.org/coding-standards/php-code.html page. 

Yes, I mean those chained things. 

In the absence of a defined standard I would go for something with the arrows aligned and no spaces before the first arrow and the object:

```php
$wa->useScript('multiselect')
   ->useScript('com_installer.changelog');
```

PHPStorm seems to do something different 

PSR-12 has nothing to say on the matter and neither does PSR-2

So I guess just pick one, and implement it consistently. 

At the moment your PR has a myriad of different examples of indentation and spaces after the initial object.

**Comment by PhilETaylor** [comment]
Other well used examples are:

```
$this->fooService
    ->doBar()
    ->doBaz();
```

```
$this
    ->fooService
    ->doBar()
    ->doBaz();
```

**Comment by brianteeman** [comment]
they should all be the same now. viewing on github always accentuates the tab display

**Comment by PhilETaylor** [comment]
Yes I see they are all the same now and using tabs.... looks horrid in the Github diff  still

Screenshot 2021-12-23 at 14 09 43


Looks wrong in phpStorm too

Screenshot 2021-12-23 at 14 09 15


but looks ok in markdown before submitting
Screenshot 2021-12-23 at 14 09 56

and after using phpStorm to "cleanup" the file it looks even more crazy
Screenshot 2021-12-23 at 14 10 22

I give up :) as long as they are consistent its not so obvious I guess. Thanks for trying.

**Comment by brianteeman** [comment]
the pr is just for using web assets and not coding standards. It was my copy/paste error that introduced an extra tab that I have now removed

**Comment by PhilETaylor** [comment]
Every PR should be about Code Standards. That's literally the whole point of standards.

**Comment by PhilETaylor** [comment]
Here is an example from Joomla 4 - which contradicts your chosen indentation (two tabs tab) for multi line chained selects and uses one tab and not two... I guess its acceptable to just do whatever... 

Screenshot 2021-12-23 at 14 17 21

**Comment by brianteeman** [comment]
sorry phil but that is exactly the same as this pr. you are getting confused by the github ui using 8spaces for a tab

File: administrator/components/com_config/tmpl/component/default.php [hunk 1]
```diff
@@ -25,10 +25,8 @@
 
 /** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
 $wa = $this->document->getWebAssetManager();
-
-// Load the tooltip behavior.
-HTMLHelper::_('behavior.formvalidator');
-HTMLHelper::_('behavior.keepalive');
+$wa->useScript('formvalidator')
```
Comment (Quy) [inline administrator/components/com_config/tmpl/component/default.php]:
```suggestion
$wa->useScript('form.validate')
```

**Comment by brianteeman** [comment]
Would be great if this could be tested and merged as there are dependent pull requests to be created afterwards

**Comment by brianteeman** [comment]
oh well another release goes by

**Comment by Fedik** [comment]
There a bug in multiselect.js, it require scriptOption.
Your changes will load multiselect.js but it will be inactive.

Need to update this:
https://github.com/joomla/joomla-cms/blob/2dc4447ee45989f905c41e9b857fe9fcb869a18d/build/media_source/system/js/multiselect.es6.js#L101-L114

To:
```javascript
  const onBoot = () => {
    let formId = '#adminForm';
    if (Joomla && Joomla.getOptions('js-multiselect', {}).formName) {
      formId = `#${Joomla.getOptions('js-multiselect', {}).formName}`;
    }
    // eslint-disable-next-line no-new
    new JMultiSelect(formId);
  };
```

And it need to be tested, at least on "articles" view, if it works there, then it will work in other places also.

**Comment by brianteeman** [comment]
Are you sure that its this change that is stopping it working?

There is already an open issue from 2019 that it didnt work https://github.com/joomla/joomla-cms/issues/25500

**Comment by brianteeman** [comment]
forgot to add I tested your change and it didnt fix it

**Comment by Fedik** [comment]
>Are you sure that its this change that is stopping it working?

Yes, I do not know all features,  but I tested simple click on the row (and console.log() for check initialization), before the patch it worked, and after the patch it stopped.
So I did code changes (that I suggested previously), to fix the `boot()` method.

**Comment by brianteeman** [comment]
I didn't observe any change in behaviour

**Comment by Fedik** [comment]
> I didn't observe any change in behaviour

Do next:
Open Articles table. Click on any row (not checkbox directly). You should see that associated checkbox checked automatically.
Then remove `HTMLHelper::_('behavior.multiselect');` and add `useScript('multiselect')` in `administrator/components/com_content/tmpl/articles/default.php`, and try the same test, now the click on the row does not do anything.

**Comment by brianteeman** [comment]
didnt know you could click on the row - will fix that in this pr with your code. Still leaves the open shift select issue

**Comment by Fedik** [comment]
> Still leaves the open shift select issue

Yeap, that seems a main point of use multiselect.js (in joomla 2 and 3), without this feature it useless and can be just trashed.
But it another issue.

**Comment by brianteeman** [comment]
Also didnt help my tests that I had made a mistake in com_content and put the change inside an if block for workflows

**Comment by brianteeman** [comment]
Multiselect is now working exactly as it was before this pr

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 1805818e4dee7fcb5bb8262b2e145ceea52745d9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36373">issues.joomla.org/tracker/joomla-cms/36373</a>.</sub>

**Comment by Quy** [comment]
Please fix conflicts.

**Comment by brianteeman** [comment]
> Please fix conflicts.

Less of a conflict and more of a "shouldnt have been merged in the first place" - fixed

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 2e7d7b676cacd85f8efabfaebdecb6e4ed13fa80<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36373">issues.joomla.org/tracker/joomla-cms/36373</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36373">issues.joomla.org/tracker/joomla-cms/36373</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Comment by brianteeman** [comment]
great thanks - now I can fix the rest

**Pull request merged**